File: CS/Form1.cs
using DevExpress.XtraRichEdit.API.Native;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace DocumentIteratorExample
{
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm
    {
        public Form1()
        {
            InitializeComponent();
            ribbonControl1.SelectedPage = pageIterator;
            richEditControl1.LoadDocument("TextWithDifferentFonts.docx");
        }

        private void btnIteratorRun_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            #region #runvisitor
            MyVisitor visitor = new MyVisitor();
            DocumentIterator iterator = new DocumentIterator(richEditControl1.Document, true);
            while (iterator.MoveNext())
                iterator.Current.Accept(visitor);
            listBoxControl1.DataSource = visitor.DocumentFonts;
            #endregion #runvisitor
        }
    }
    #region #myvisitorclass
    public class MyVisitor : DocumentVisitorBase
    {
        private List<string> fonts;
        public List<string> DocumentFonts {get {return fonts;}}
        public MyVisitor()
        {
            fonts = new List<string>();
        }

        public override void Visit(DocumentText text) {
            if (!fonts.Contains(text.TextProperties.FontName))
                fonts.Add(text.TextProperties.FontName);
        }
    }
    #endregion #myvisitorclass
}

File: VB/Form1.vb
Imports DevExpress.XtraRichEdit.API.Native
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace DocumentIteratorExample
    Partial Public Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Public Sub New()
            InitializeComponent()
            ribbonControl1.SelectedPage = pageIterator
            richEditControl1.LoadDocument("TextWithDifferentFonts.docx")
        End Sub

        Private Sub btnIteratorRun_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles btnIteratorRun.ItemClick
'            #Region "#runvisitor"
            Dim visitor As New MyVisitor()
            Dim [iterator] As New DocumentIterator(richEditControl1.Document, True)
            Do While [iterator].MoveNext()
                [iterator].Current.Accept(visitor)
            Loop
            listBoxControl1.DataSource = visitor.DocumentFonts
'            #End Region ' #runvisitor
        End Sub
    End Class
    #Region "#myvisitorclass"
    Public Class MyVisitor
        Inherits DocumentVisitorBase

        Private _Fonts As List(Of String)

        Public ReadOnly Property DocumentFonts() As List(Of String)
            Get
                Return _Fonts
            End Get
        End Property

        Public Sub New()
            _Fonts = New List(Of String)()
        End Sub

        Public Overrides Sub Visit(ByVal text As DocumentText)
            If Not _Fonts.Contains(text.TextProperties.FontName) Then
                _Fonts.Add(text.TextProperties.FontName)
            End If
        End Sub
    End Class
    #End Region ' #myvisitorclass
End Namespace
